orthoinference/src/main/java/org/reactome/orthoinference/EventsInferrer.java

```diff
@@ -165,19 +168,19 @@ public static void inferEvents(Properties props, String species) throws Exceptio
 			logger.info("Attempting RlE inference: " + reactionInst);
 			// Check if the current Reaction already exists for this species, that it is a valid instance (passes some filters), and that it doesn't have a Disease attribute.
 			// Adds to manualHumanEvents array if it passes conditions. This code block allows you to re-run the code without re-inferring instances.
-			List<GKInstance> previouslyInferredInstances = new ArrayList<GKInstance>();
+			List<GKInstance> previouslyInferredInstances = new ArrayList<>();
 			previouslyInferredInstances = checkIfPreviouslyInferred(reactionInst, orthologousEvent, previouslyInferredInstances);
-			previouslyInferredInstances = checkIfPreviouslyInferred(reactionInst, inferredFrom, previouslyInferredInstances);
-			if (previouslyInferredInstances.size() > 0)
-			{
+			previouslyInferredInstances.addAll(checkIfPreviouslyInferred(reactionInst, inferredFrom, previouslyInferredInstances));
+			if (previouslyInferredInstances.size() > 0) {
 				GKInstance prevInfInst = previouslyInferredInstances.get(0);
-				if (prevInfInst.getAttributeValue(disease) == null)
-				{
+				GKInstance prevInfSummationInst = (GKInstance) prevInfInst.getAttributeValue(summation);
+				String prevInfSummationText = prevInfSummationInst.getAttributeValue(text).toString();
+				if (prevInfInst.getAttributeValue(disease) == null &&  prevInfSummationText.equals(summationText)) {
 					logger.info("Inferred RlE already exists, skipping inference");
-					manualEventToNonHumanSource.put(reactionInst, prevInfInst);
-					manualHumanEvents.add(reactionInst);
+					eventsAlreadyInferredMap.put(reactionInst, prevInfInst);
+					eventsAlreadyInferred.add(reactionInst);
```

Comment:
Are the `eventsAlreadyInferredMap` and `eventsAlreadyInferred` data structures both needed?  Could the list populated in `eventsAlreadyInferred` be obtained from the key set of the `eventsAlreadyInferredMap`?